### PR TITLE
[MERGE] GE04 Module Finished

### PR DESCRIPTION
**Pull request by ivgm-odoo**
### Features

- Pricelist with $2500 discount
- Button in sales order, to apply the pricelist
- Discount to new purchase motorcycle

### CODE REVIEW

File: ge04_team2/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+    "name": "Motorcycle Discount",
+    "summary": "Manage Registration of Motorcycles",
+    "description": """
+    Motorcycle Discount
+====================
+This Module is used to show a discount when a new customer purchase a motorcycle in them first time.
```
Comment (ayhp-odoo) [inline ge04_team2/__manifest__.py:7]: Please be more explicit what does this module do

File: ge04_team2/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "version": "0.1",
+    "category": "Kauil/Discount",
+    "license": "OPL-1",
+    "depends": ["sale_management"],
+
+    "data": [
+        'data/discount_data.xml',
+        'views/sale_order_inherit.xml'
+    ],
+    "demo": [],
```
Comment (ayhp-odoo) [inline ge04_team2/__manifest__.py:18]: it is empty please remove it

File: ge04_team2/models/ge04_team2.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+from odoo import api, fields, models
+
+
+class ge04Team2(models.Model):
```
Comment (ayhp-odoo) [inline ge04_team2/models/ge04_team2.py:4]: The name of the class should be the same as the one you are inheriting

File: ge04_team2/models/ge04_team2.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    _inherit = 'sale.order'
+
+    is_new_customer = fields.Boolean(compute="_compute_first_purchase")
+
+    @api.depends('partner_id')
+    def _compute_first_purchase(self):
```
Comment (ayhp-odoo) [inline ge04_team2/models/ge04_team2.py:10]: The compute method of a field should just assign the value for the field, please separate the extra logic into another function

File: ge04_team2/models/ge04_team2.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            partner = record.partner_id.id
+            sale_orders = self.env['sale.order'].search(
+                [('partner_id', '=', partner)])
+
+            order_lines = sale_orders.filtered(
+                lambda r: r.type_name == 'Sales Order').mapped('order_line')
+
+            products = order_lines.mapped('product_id')
+
+            print(products)
```
Comment (ayhp-odoo) [inline ge04_team2/models/ge04_team2.py:23]: remove prints, please look for another ways to debug

File: ge04_team2/data/discount_data.xml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+        <record id="discount_first_purchase" model="product.pricelist">
+            <field name="name">Discount first purchase</field> 
+        </record>
+        <record id="items_discount_first_purchase" model="product.pricelist.item">
+            <field name="pricelist_id" ref="discount_first_purchase"/>
+            <field name="compute_price">formula</field>
+            <field name="price_surcharge">-2500</field>
+        </record>
+    </data>
+</odoo>
```
Comment (ayhp-odoo) [inline ge04_team2/data/discount_data.xml:14]: new line at the end of the file and the other ones where is missing

File: ge04_team2/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
+{
+    "name": "Motorcycle Discount",
+    "summary": "Manage Registration of Motorcycles",
+    "description": """
+    Motorcycle Discount
+====================
+This Module is used to show a discount when a new customer purchase a motorcycle in them first time.
+    """,
+    "version": "0.1",
```
Comment (ayhp-odoo) [inline ge04_team2/__manifest__.py:9]: Version should be 1.0.0

File: ge04_team2/views/sale_order_inherit.xml [hunk 8]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_order_form_inherit_discount" model="ir.ui.view">
+        <field name="name">sale.order.form.inherit.discount</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form"/>
+        <field name="arch" type="xml">
+<!--             Use Xpath -->
```
Comment (ayhp-odoo) [inline ge04_team2/views/sale_order_inherit.xml:8]: remove unnecessary comments

**Review by ayhp-odoo — CHANGES_REQUESTED** [review]
Please check Odoo's git guidelines for commits and pull requests, you need to explain what does this module will do.

https://www.odoo.com/documentation/16.0/contributing/development/git_guidelines.html?highlight=guidelines#configure-your-git

In the models directory the files name should be the same as the class, please refer to the standard code

File: ge04_team2/models/ge04_team2.py (hunk 3 above)
Comment (ivgm-odoo) [reply ge04_team2/models/ge04_team2.py]: Solved

File: ge04_team2/models/ge04_team2.py (hunk 4 above)
Comment (ivgm-odoo) [reply ge04_team2/models/ge04_team2.py]: Solved

File: ge04_team2/models/sale_order.py [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    is_new_customer = fields.Boolean(compute="_compute_first_purchase")
+
+    def _calculate_is_new_customer(self, customer):
+        is_new_customer = True
+        partner = customer.partner_id.id
+        sale_orders = self.env['sale.order'].search(
+            [('partner_id', '=', partner)])
+        order_lines = sale_orders.filtered(
+            lambda r: r.type_name == 'Sales Order').mapped('order_line')
+        products = order_lines.mapped('product_id')
```
Comment (ayhp-odoo) [inline ge04_team2/models/sale_order.py:16]: Better do in this way:

`order_lines.mapped('product_id').filtered(lambda product: product.detailed_type == 'motorcycle').write('is_new_customer': False})`

File: ge04_team2/models/sale_order.py [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        for i in products:
```
Comment (ayhp-odoo) [inline ge04_team2/models/sale_order.py:17]: delete this for is not necessary anymore

File: ge04_team2/models/sale_order.py [hunk 11]
```diff
@@ -0,0 +1,35 @@
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    is_new_customer = fields.Boolean(compute="_compute_first_purchase")
+
+    def _calculate_is_new_customer(self, customer):
+        is_new_customer = True
```
Comment (ayhp-odoo) [inline ge04_team2/models/sale_order.py:10]: delete this and partner, partner you can put it on the search

File: ge04_team2/models/sale_order.py [hunk 12]
```diff
@@ -0,0 +1,35 @@
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    is_new_customer = fields.Boolean(compute="_compute_first_purchase")
+
+    def _calculate_is_new_customer(self, customer):
```
Comment (ayhp-odoo) [inline ge04_team2/models/sale_order.py:9]: move everything else to the compute and delete this function, due we reduce the logic is possible now to go in the compute

File: ge04_team2/data/discount_data.xml (hunk 6 above)
Comment (ayhp-odoo) [inline ge04_team2/data/discount_data.xml:14]: New line

**Review by ayhp-odoo — CHANGES_REQUESTED** [review]

**Review by ayhp-odoo — APPROVED** [review]

**Pull request merged**